src/Zend/Loader/ClassMapAutoloader.php

```diff
@@ -206,7 +206,13 @@ public static function realPharPath($path)
         $parts = explode('/', str_replace(array('/','\\'), '/', substr($path, 8)));
         $parts = array_values(array_filter($parts, array(__CLASS__, 'concatPharParts')));
 
-        array_walk($parts, array(__CLASS__, 'resolvePharParentPath'), $parts);
+        array_walk($parts, function ($value, $key) use (&$parts) {
```

Comment:
PHP 8 threw an error here because the third parameter of the specified method is defined as pass-by-reference, but array_walk doesn't pass that value by reference (perhaps PHP once did, but seems like it hasn't in a long time (https://www.php.net/manual/en/function.array-walk.php#56897 I should note that the approach used in the linked comment (specifying the pass-by-reference at the call site) isn't possible with today's PHP). This closure should behave the same as the method approach was intended, but the tests don't actually test this case (I couldn't get this to cause test failures at all).

Reply:
This is fine, but is it not possible to call `self::resolvePharParentPath` from the closure?

Reply:
Ah, yes, that would be better (updated)